addons/sys_logistics/fnc_setObjectCargo.sqf

```diff
@@ -70,19 +66,33 @@ _typesWeapons = [[_cargoW,"WeaponCargo"],[_cargoM,"MagazineCargo"],[_cargoI,"Ite
 
     _content = _x select 0;
     _operation = _x select 1;
-    _current = call (compile format["get%1 _input",_operation]);
 
-    if !(str(_content) == str(_current)) then {
+    _currenttemp = [{getWeaponCargo _input}, {getMagazineCargo _input},{getItemCargo _input}];
+    _current = _currenttemp select (["weaponcargo","magazinecargo","itemcargo"] find (tolower _operation));
+    call _current;
 
-        if (count (_current select 0) > 0) then {call (compile format["clear%1%2 _input",_operation,_global])};
+    if !(_content isEqualTo _current) then {
+
+        if (count (_current select 0) > 0) then {
+
+            _clear = [[{clearWeaponCargo _input}, {clearMagazineCargo _input},{clearItemCargo _input}],
+                [{clearWeaponCargoGlobal _input}, {clearMagazineCargoGlobal _input},{clearItemCargoGlobal _input}]];   
+            _clearloc = _clear select (_global = "Global");
+            _clearcommand = _clearloc select (["weaponcargo","magazinecargo","itemcargo"] find (tolower _operation));
+            call _clearcommand;
```

Comment:
Though, the comment above about the reset thingy probably applies here as well.

Reply:
Suggestion: ```sqf
        if (count (_current select 0) > 0) then {
            
            if (_global == "Global") then {
                _actions = [{clearWeaponCargoGlobal _input}, {clearMagazineCargoGlobal _input},{clearItemCargoGlobal _input}]; }
                else {
                _actions = [{clearWeaponCargo _input}, {clearMagazineCargo _input},{clearItemCargo _input}];
                };
            };

            call ( _actions select (["weaponcargo","magazinecargo","itemcargo"] find (tolower _operation)) );
        };
``` Reset weapons and items state

Reply:
Seems good to me. There are probably better solutions to the whole thing but at least it's a bit more readable. :)

Also there's a `}` too many  on line 4 in your snippet at the end.

Reply:
Yea I cleaned up the code a bit after the complete implementation.

Reply:
Full change in latest commit to branch: https://github.com/TheNightstalk3r/ALiVE.OS/commit/5d779ec528230b6589e2f4e0d314e6c4739fb4bf